WcaOnRails/app/views/competitions/_nearby_competitions.html.erb

```diff
@@ -11,6 +11,9 @@
   <div class="col-sm-9">
     <% if @competition.has_date? && @competition.has_location? %>
       <% nearby_competitions = @competition.nearby_competitions(Competition::NEARBY_DAYS_WARNING, Competition::NEARBY_DISTANCE_KM_WARNING)[0, 10] %>
+      <% if !current_user.can_view_hidden_competitions? %>
```

Comment:
I'm not sure how to do the alert stuff at the controller. But I agree that at least the nearby_competitions object can be defined there.